src/main/kotlin/graphics/scenery/primitives/Atmosphere.kt

```diff
@@ -17,51 +17,66 @@ import kotlin.math.*
  * @param latitude Latitude of the user; needed to calculate the local sun position. Defaults to 50.0, which is central Germany.
  */
-open class Atmosphere(initSunDir: Vector3f? = null, emissionStrength: Float = 0.3f, var latitude: Double = 50.0) :
+open class Atmosphere(
+    initSunDirection: Vector3f? = null,
+    var emissionStrength: Float = 1f,
+    var latitude: Float = 50.0f
+) :
     Icosphere(10f, 2, insideNormals = true) {
 
     @ShaderProperty
-    var sunDir: Vector3f
+    var sunDirection = Vector3f(1f, 1f, 1f)
 
+    /** Is set to true if the user manually moved the sun direction. This disables automatic updating.*/
     private var sunDirectionManual: Boolean = false
+    /** Flag that tracks whether the sun position controls are currently attached to the input handler. */
+    var hasControls: Boolean = false
+
+    // Coroutine job for updating the sun direction
+    var job: Job
 
     init {
         this.name = "Atmosphere"
         setMaterial(ShaderMaterial.fromClass(this::class.java))
         material {
             cullingMode = Material.CullingMode.Front
             depthTest = Material.DepthTest.LessEqual
-            emissive = Vector4f(0f, 0f, 0f, emissionStrength)
+            emissive = Vector4f(0f, 0f, 0f, emissionStrength * 0.3f)
         }
 
         // Only use time-based elevation when the formal parameter is empty
-        if (initSunDir == null) {
-            sunDir = getSunDirFromTime()
+        if (initSunDirection == null) {
+            getSunDirectionFromTime()
         }
         else {
-            sunDir = initSunDir
+            sunDirection = initSunDirection
             sunDirectionManual = true
         }
 
         // Spawn a coroutine to update the sun direction
-        val job = CoroutineScope(Dispatchers.Default).launch {
+        job = CoroutineScope(Dispatchers.Default).launch {
             while (!sunDirectionManual) {
-                sunDir = getSunDirFromTime()
-                // Wait 30 seconds
-                delay(30 * 1000)
+                getSunDirectionFromTime()
+                // Wait 10 seconds
+                delay(10 * 1000)
             }
         }
     }
 
-    /** Turn the current local time into a sun elevation angle, encoded as cartesian  [Vector3f].
+    /** Rewrite the atmosphere material with the (potentially) changed emission value stored in [emissionStrength] */
+    fun updateEmissionStrength() {
+        material { emissive = Vector4f(0f, 0f, 0f, emissionStrength * 0.3f) }
+    }
+
+    /** Turn the current local time into a sun elevation angle, encoded as cartesian.
      * @param localTime local time parameter, defaults to [LocalDateTime.now].
      */
-    private fun getSunDirFromTime(localTime: LocalDateTime = LocalDateTime.now()): Vector3f {
-        val latitudeRad = toRadians(latitude)
+    fun getSunDirectionFromTime(localTime: LocalDateTime = LocalDateTime.now()) {
```

Comment:
The name is deceiving - shouldn't it rather be called `setSunDirectionFromTime`, as it's setting the direction?

Reply:
Alternatively, keep it like this, but let it only return the direction, _and_ add a function that calls this one but also sets the direction.

Reply:
No I think you are right. It used to be a getter, but then I changed the method to directly write into `sunDirection` and forgot to update the method name.